### PR TITLE
Reduces small allocations on heap

### PR DESCRIPTION
**Pull request by gdavison**
### Description

Reduces small allocations on heap so that they don't have to be garbage collected later. In generated tagging and service endpoint tests as well as `acctest`, `provider` and `xray` service.

### CODE REVIEW

**Review by YakDriver — APPROVED** [review]
Interestingly, this resulted in a 0.08% reduction in max RSS based on 3 runs with and without the changes. I just wanted to make sure it didn't increase but a little surprised at a reduction.

```console
% make sane
make: Sane Check (48 tests of Top 30 resources)
make: Like 'sanity' except full output and stops soon after 1st error
make: NOTE: NOT an exhaustive set of tests! Finds big problems only.
2024/12/18 14:51:55 Initializing Terraform AWS Provider...
=== RUN   TestAccIAMInstanceProfile_tags
=== PAUSE TestAccIAMInstanceProfile_tags
=== RUN   TestAccIAMInstanceProfile_tags_null
=== PAUSE TestAccIAMInstanceProfile_tags_null
=== RUN   TestAccIAMInstanceProfile_tags_EmptyMap
=== PAUSE TestAccIAMInstanceProfile_tags_EmptyMap
=== RUN   TestAccIAMInstanceProfile_tags_AddOnUpdate
=== PAUSE TestAccIAMInstanceProfile_tags_AddOnUpdate
=== RUN   TestAccIAMInstanceProfile_tags_EmptyTag_OnCreate
=== PAUSE TestAccIAMInstanceProfile_tags_EmptyTag_OnCreate
=== RUN   TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_providerOnly
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_providerOnly
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_nonOverlapping
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_overlapping
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_overlapping
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccIAMInstanceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccIAMInstanceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccIAMInstanceProfile_tags_ComputedTag_OnCreate
=== PAUSE TestAccIAMInstanceProfile_tags_ComputedTag_OnCreate
=== RUN   TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== PAUSE TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== RUN   TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_ResourceTag
=== PAUSE TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_ResourceTag
=== RUN   TestAccIAMInstanceProfile_basic
=== PAUSE TestAccIAMInstanceProfile_basic
=== RUN   TestAccIAMPolicyDocumentDataSource_basic
=== PAUSE TestAccIAMPolicyDocumentDataSource_basic
=== RUN   TestAccIAMPolicyDocumentDataSource_sourceConflicting
=== PAUSE TestAccIAMPolicyDocumentDataSource_sourceConflicting
=== RUN   TestAccIAMPolicy_tags
=== PAUSE TestAccIAMPolicy_tags
=== RUN   TestAccIAMPolicy_tags_null
=== PAUSE TestAccIAMPolicy_tags_null
=== RUN   TestAccIAMPolicy_tags_EmptyMap
=== PAUSE TestAccIAMPolicy_tags_EmptyMap
=== RUN   TestAccIAMPolicy_tags_AddOnUpdate
=== PAUSE TestAccIAMPolicy_tags_AddOnUpdate
=== RUN   TestAccIAMPolicy_tags_EmptyTag_OnCreate
=== PAUSE TestAccIAMPolicy_tags_EmptyTag_OnCreate
=== RUN   TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccIAMPolicy_tags_DefaultTags_providerOnly
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_providerOnly
=== RUN   TestAccIAMPolicy_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_nonOverlapping
=== RUN   TestAccIAMPolicy_tags_DefaultTags_overlapping
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_overlapping
=== RUN   TestAccIAMPolicy_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccIAMPolicy_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccIAMPolicy_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccIAMPolicy_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccIAMPolicy_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccIAMPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccIAMPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccIAMPolicy_tags_ComputedTag_OnCreate
=== PAUSE TestAccIAMPolicy_tags_ComputedTag_OnCreate
=== RUN   TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccIAMPolicy_tags_IgnoreTags_Overlap_DefaultTag
=== PAUSE TestAccIAMPolicy_tags_IgnoreTags_Overlap_DefaultTag
=== RUN   TestAccIAMPolicy_tags_IgnoreTags_Overlap_ResourceTag
=== PAUSE TestAccIAMPolicy_tags_IgnoreTags_Overlap_ResourceTag
=== RUN   TestAccIAMPolicy_basic
=== PAUSE TestAccIAMPolicy_basic
=== RUN   TestAccIAMPolicy_policy
=== PAUSE TestAccIAMPolicy_policy
=== RUN   TestAccIAMPolicy_policyDuplicateKeys
=== PAUSE TestAccIAMPolicy_policyDuplicateKeys
=== RUN   TestAccIAMRolePolicyAttachment_basic
=== PAUSE TestAccIAMRolePolicyAttachment_basic
=== RUN   TestAccIAMRolePolicyAttachment_disappears
=== PAUSE TestAccIAMRolePolicyAttachment_disappears
=== RUN   TestAccIAMRolePolicyAttachment_Disappears_role
=== PAUSE TestAccIAMRolePolicyAttachment_Disappears_role
=== RUN   TestAccIAMRolePolicy_basic
=== PAUSE TestAccIAMRolePolicy_basic
=== RUN   TestAccIAMRolePolicy_unknownsInPolicy
=== PAUSE TestAccIAMRolePolicy_unknownsInPolicy
=== RUN   TestAccIAMRole_basic
=== PAUSE TestAccIAMRole_basic
=== RUN   TestAccIAMRole_namePrefix
=== PAUSE TestAccIAMRole_namePrefix
=== RUN   TestAccIAMRole_disappears
=== PAUSE TestAccIAMRole_disappears
=== RUN   TestAccIAMRole_InlinePolicy_basic
=== PAUSE TestAccIAMRole_InlinePolicy_basic
=== CONT  TestAccIAMInstanceProfile_tags
=== CONT  TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Add
=== CONT  TestAccIAMPolicy_tags_IgnoreTags_Overlap_DefaultTag
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== CONT  TestAccIAMPolicy_tags_EmptyTag_OnCreate
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_nullOverlappingResourceTag
=== CONT  TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_updateToResourceOnly
=== CONT  TestAccIAMInstanceProfile_basic
=== CONT  TestAccIAMPolicy_tags_DefaultTags_emptyResourceTag
=== CONT  TestAccIAMPolicy_tags_ComputedTag_OnCreate
=== CONT  TestAccIAMPolicy_tags_DefaultTags_nullOverlappingResourceTag
=== CONT  TestAccIAMPolicyDocumentDataSource_basic
=== CONT  TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Add
=== CONT  TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Replace
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_providerOnly
=== CONT  TestAccIAMPolicy_tags_DefaultTags_emptyProviderOnlyTag
=== CONT  TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Replace
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_emptyProviderOnlyTag
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_emptyResourceTag
=--- PASS: TestAccIAMPolicyDocumentDataSource_basic (12.62s)
=== CONT  TestAccIAMRolePolicyAttachment_Disappears_role
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nullOverlappingResourceTag (22.68s)
=== CONT  TestAccIAMRole_InlinePolicy_basic
--- PASS: TestAccIAMPolicy_tags_DefaultTags_emptyProviderOnlyTag (23.60s)
=== CONT  TestAccIAMRole_disappears
--- PASS: TestAccIAMPolicy_tags_DefaultTags_emptyResourceTag (23.98s)
=== CONT  TestAccIAMRole_namePrefix
--- PASS: TestAccIAMPolicy_tags_ComputedTag_OnCreate (24.69s)
=== CONT  TestAccIAMRole_basic
--- PASS: TestAccIAMInstanceProfile_basic (29.33s)
=== CONT  TestAccIAMRolePolicy_unknownsInPolicy
--- PASS: TestAccIAMRolePolicyAttachment_Disappears_role (18.45s)
=== CONT  TestAccIAMRolePolicy_basic
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_emptyResourceTag (32.58s)
=== CONT  TestAccIAMInstanceProfile_tags_EmptyTag_OnCreate
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_emptyProviderOnlyTag (33.33s)
=== CONT  TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nullOverlappingResourceTag (33.53s)
=== CONT  TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Add
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nullNonOverlappingResourceTag (33.93s)
=== CONT  TestAccIAMInstanceProfile_tags_ComputedTag_OnCreate
--- PASS: TestAccIAMRole_disappears (17.10s)
=== CONT  TestAccIAMPolicy_basic
--- PASS: TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Replace (41.25s)
=== CONT  TestAccIAMPolicy_policy
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnCreate (41.29s)
=== CONT  TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_ResourceTag
--- PASS: TestAccIAMRole_namePrefix (19.31s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
--- PASS: TestAccIAMRole_basic (19.46s)
=== CONT  TestAccIAMInstanceProfile_tags_EmptyMap
--- PASS: TestAccIAMPolicy_tags_IgnoreTags_Overlap_DefaultTag (47.11s)
=== CONT  TestAccIAMInstanceProfile_tags_AddOnUpdate
--- PASS: TestAccIAMRolePolicy_unknownsInPolicy (21.71s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_overlapping
--- PASS: TestAccIAMRolePolicy_basic (20.69s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_updateToResourceOnly (52.36s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Replace (55.03s)
=== CONT  TestAccIAMInstanceProfile_tags_null
--- PASS: TestAccIAMInstanceProfile_tags_ComputedTag_OnUpdate_Add (56.72s)
=== CONT  TestAccIAMRolePolicyAttachment_disappears
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Add (57.64s)
=== CONT  TestAccIAMPolicy_tags_IgnoreTags_Overlap_ResourceTag
--- PASS: TestAccIAMPolicy_basic (21.43s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_providerOnly
--- PASS: TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_DefaultTag (62.24s)
=== CONT  TestAccIAMPolicy_tags_DefaultTags_nonOverlapping
--- PASS: TestAccIAMInstanceProfile_tags_ComputedTag_OnCreate (31.11s)
=== CONT  TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nullNonOverlappingResourceTag (23.86s)
=== CONT  TestAccIAMPolicy_tags_null
--- PASS: TestAccIAMRole_InlinePolicy_basic (46.49s)
=== CONT  TestAccIAMPolicy_tags_AddOnUpdate
--- PASS: TestAccIAMRolePolicyAttachment_disappears (18.99s)
=== CONT  TestAccIAMPolicy_tags_EmptyMap
--- PASS: TestAccIAMPolicy_policy (35.23s)
=== CONT  TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Add
--- PASS: TestAccIAMInstanceProfile_tags_EmptyMap (37.50s)
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_overlapping
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Replace (53.20s)
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnCreate (57.89s)
=== CONT  TestAccIAMInstanceProfile_tags_DefaultTags_nonOverlapping
--- PASS: TestAccIAMPolicy_tags_DefaultTags_updateToResourceOnly (40.42s)
=== CONT  TestAccIAMPolicy_tags
--- PASS: TestAccIAMPolicy_tags_DefaultTags_updateToProviderOnly (41.29s)
=== CONT  TestAccIAMRolePolicyAttachment_basic
--- PASS: TestAccIAMInstanceProfile_tags_null (38.94s)
=== CONT  TestAccIAMPolicyDocumentDataSource_sourceConflicting
--- PASS: TestAccIAMInstanceProfile_tags_AddOnUpdate (50.80s)
=== CONT  TestAccIAMPolicy_policyDuplicateKeys
--- PASS: TestAccIAMPolicy_tags_null (31.96s)
--- PASS: TestAccIAMPolicy_policyDuplicateKeys (1.61s)
--- PASS: TestAccIAMPolicy_tags_EmptyMap (28.63s)
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Replace (40.75s)
--- PASS: TestAccIAMPolicyDocumentDataSource_sourceConflicting (13.30s)
--- PASS: TestAccIAMPolicy_tags_AddOnUpdate (40.77s)
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Add (77.37s)
--- PASS: TestAccIAMInstanceProfile_tags (111.48s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_providerOnly (114.40s)
--- PASS: TestAccIAMInstanceProfile_tags_IgnoreTags_Overlap_ResourceTag (73.74s)
--- PASS: TestAccIAMPolicy_tags_IgnoreTags_Overlap_ResourceTag (58.49s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_overlapping (65.18s)
--- PASS: TestAccIAMPolicy_tags_ComputedTag_OnUpdate_Add (40.51s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nonOverlapping (59.82s)
--- PASS: TestAccIAMRolePolicyAttachment_basic (28.84s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_updateToProviderOnly (42.47s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_providerOnly (71.31s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_overlapping (64.78s)
--- PASS: TestAccIAMPolicy_tags (56.10s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nonOverlapping (60.61s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/iam	155.166s
2024/12/18 14:55:01 Initializing Terraform AWS Provider...
=== RUN   TestAccLogsGroup_basic
=== PAUSE TestAccLogsGroup_basic
=== RUN   TestAccLogsGroup_multiple
=== PAUSE TestAccLogsGroup_multiple
=== CONT  TestAccLogsGroup_basic
=== CONT  TestAccLogsGroup_multiple
--- PASS: TestAccLogsGroup_multiple (12.20s)
--- PASS: TestAccLogsGroup_basic (13.86s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/logs	18.022s
2024/12/18 14:55:05 Initializing Terraform AWS Provider...
=== RUN   TestAccVPCDataSource_basic
=== PAUSE TestAccVPCDataSource_basic
=== RUN   TestAccVPCRouteTableAssociation_Subnet_basic
=== PAUSE TestAccVPCRouteTableAssociation_Subnet_basic
=== RUN   TestAccVPCRouteTable_basic
=== PAUSE TestAccVPCRouteTable_basic
=== RUN   TestAccVPCSecurityGroupRule_race
=== PAUSE TestAccVPCSecurityGroupRule_race
=== RUN   TestAccVPCSecurityGroupRule_protocolChange
=== PAUSE TestAccVPCSecurityGroupRule_protocolChange
=== RUN   TestAccVPCSecurityGroup_basic
=== PAUSE TestAccVPCSecurityGroup_basic
=== RUN   TestAccVPCSecurityGroup_egressMode
=== PAUSE TestAccVPCSecurityGroup_egressMode
=== RUN   TestAccVPCSecurityGroup_vpcAllEgress
=== PAUSE TestAccVPCSecurityGroup_vpcAllEgress
=== RUN   TestAccVPCSubnet_basic
=== PAUSE TestAccVPCSubnet_basic
=== RUN   TestAccVPC_tenancy
=== PAUSE TestAccVPC_tenancy
=== CONT  TestAccVPCDataSource_basic
=== CONT  TestAccVPCSecurityGroup_basic
=== CONT  TestAccVPCSecurityGroupRule_protocolChange
=== CONT  TestAccVPC_tenancy
=== CONT  TestAccVPCSecurityGroup_vpcAllEgress
=== CONT  TestAccVPCRouteTable_basic
=== CONT  TestAccVPCSecurityGroup_egressMode
=== CONT  TestAccVPCRouteTableAssociation_Subnet_basic
=== CONT  TestAccVPCSecurityGroupRule_race
=== CONT  TestAccVPCSubnet_basic
--- PASS: TestAccVPCSubnet_basic (20.22s)
--- PASS: TestAccVPCRouteTable_basic (20.70s)
--- PASS: TestAccVPCSecurityGroup_basic (21.79s)
--- PASS: TestAccVPCSecurityGroup_vpcAllEgress (23.11s)
--- PASS: TestAccVPCRouteTableAssociation_Subnet_basic (24.08s)
--- PASS: TestAccVPCDataSource_basic (29.32s)
--- PASS: TestAccVPCSecurityGroup_egressMode (42.79s)
--- PASS: TestAccVPC_tenancy (44.11s)
--- PASS: TestAccVPCSecurityGroupRule_protocolChange (51.75s)
--- PASS: TestAccVPCSecurityGroupRule_race (167.44s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	174.926s
2024/12/18 14:55:14 Initializing Terraform AWS Provider...
=== RUN   TestAccECSService_basic
=== PAUSE TestAccECSService_basic
=== RUN   TestAccECSService_basicImport
=== PAUSE TestAccECSService_basicImport
=== RUN   TestAccECSTaskDefinition_basic
=== PAUSE TestAccECSTaskDefinition_basic
=== CONT  TestAccECSService_basic
=== CONT  TestAccECSTaskDefinition_basic
=== CONT  TestAccECSService_basicImport
--- PASS: TestAccECSTaskDefinition_basic (21.61s)
--- PASS: TestAccECSService_basic (68.29s)
--- PASS: TestAccECSService_basicImport (194.41s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ecs	210.004s
2024/12/18 14:55:18 Initializing Terraform AWS Provider...
=== RUN   TestAccELBV2TargetGroup_basic
=== PAUSE TestAccELBV2TargetGroup_basic
=== CONT  TestAccELBV2TargetGroup_basic
--- PASS: TestAccELBV2TargetGroup_basic (19.81s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/elbv2	39.560s
2024/12/18 14:55:09 Initializing Terraform AWS Provider...
=== RUN   TestAccKMSKey_basic
=== PAUSE TestAccKMSKey_basic
=== CONT  TestAccKMSKey_basic
--- PASS: TestAccKMSKey_basic (22.73s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/kms	34.256s
2024/12/18 14:59:00 Initializing Terraform AWS Provider...
=== RUN   TestAccLambdaFunction_basic
=== PAUSE TestAccLambdaFunction_basic
=== RUN   TestAccLambdaPermission_basic
=== PAUSE TestAccLambdaPermission_basic
=== CONT  TestAccLambdaFunction_basic
=== CONT  TestAccLambdaPermission_basic
--- PASS: TestAccLambdaPermission_basic (29.16s)
--- PASS: TestAccLambdaFunction_basic (40.24s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lambda	52.589s
2024/12/18 14:58:56 Initializing Terraform AWS Provider...
=== RUN   TestAccMetaPartitionDataSource_basic
=== PAUSE TestAccMetaPartitionDataSource_basic
=== RUN   TestAccMetaRegionDataSource_basic
=== PAUSE TestAccMetaRegionDataSource_basic
=== RUN   TestAccMetaRegionDataSource_endpoint
=== PAUSE TestAccMetaRegionDataSource_endpoint
=== RUN   TestAccMetaRegionDataSource_endpointAndName
=== PAUSE TestAccMetaRegionDataSource_endpointAndName
=== CONT  TestAccMetaPartitionDataSource_basic
=== CONT  TestAccMetaRegionDataSource_endpoint
=== CONT  TestAccMetaRegionDataSource_endpointAndName
=== CONT  TestAccMetaRegionDataSource_basic
--- PASS: TestAccMetaRegionDataSource_basic (7.68s)
--- PASS: TestAccMetaPartitionDataSource_basic (7.69s)
--- PASS: TestAccMetaRegionDataSource_endpointAndName (8.95s)
--- PASS: TestAccMetaRegionDataSource_endpoint (8.97s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/meta	17.195s
2024/12/18 14:58:52 Initializing Terraform AWS Provider...
=== RUN   TestAccRoute53Record_basic
=== PAUSE TestAccRoute53Record_basic
=== RUN   TestAccRoute53Record_Latency_basic
=== PAUSE TestAccRoute53Record_Latency_basic
=== RUN   TestAccRoute53ZoneDataSource_name
=== PAUSE TestAccRoute53ZoneDataSource_name
=== RUN   TestAccRoute53ZoneDataSource_name_idEmptyString
=== PAUSE TestAccRoute53ZoneDataSource_name_idEmptyString
=== CONT  TestAccRoute53Record_basic
=== CONT  TestAccRoute53ZoneDataSource_name
=== CONT  TestAccRoute53Record_Latency_basic
=== CONT  TestAccRoute53ZoneDataSource_name_idEmptyString
--- PASS: TestAccRoute53ZoneDataSource_name (86.90s)
--- PASS: TestAccRoute53ZoneDataSource_name_idEmptyString (92.84s)
--- PASS: TestAccRoute53Record_basic (164.91s)
--- PASS: TestAccRoute53Record_Latency_basic (174.82s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/route53	179.057s
2024/12/18 14:59:08 Initializing Terraform AWS Provider...
=== RUN   TestAccS3BucketACL_updateACL
=== PAUSE TestAccS3BucketACL_updateACL
=== RUN   TestAccS3BucketPolicy_basic
=== PAUSE TestAccS3BucketPolicy_basic
=== RUN   TestAccS3BucketPublicAccessBlock_basic
=== PAUSE TestAccS3BucketPublicAccessBlock_basic
=== RUN   TestAccS3Bucket_Basic_basic
=== PAUSE TestAccS3Bucket_Basic_basic
=== RUN   TestAccS3Bucket_Security_corsUpdate
=== PAUSE TestAccS3Bucket_Security_corsUpdate
=== CONT  TestAccS3BucketACL_updateACL
=== CONT  TestAccS3Bucket_Basic_basic
=== CONT  TestAccS3BucketPublicAccessBlock_basic
=== CONT  TestAccS3BucketPolicy_basic
=== CONT  TestAccS3Bucket_Security_corsUpdate
--- PASS: TestAccS3Bucket_Basic_basic (18.37s)
--- PASS: TestAccS3BucketPolicy_basic (18.64s)
--- PASS: TestAccS3BucketPublicAccessBlock_basic (18.69s)
--- PASS: TestAccS3BucketACL_updateACL (31.10s)
--- PASS: TestAccS3Bucket_Security_corsUpdate (32.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/s3	51.023s
2024/12/18 14:59:12 Initializing Terraform AWS Provider...
=== RUN   TestAccSecretsManagerSecret_basic
=== PAUSE TestAccSecretsManagerSecret_basic
=== RUN   TestAccSecretsManagerSecret_basicReplica
=== PAUSE TestAccSecretsManagerSecret_basicReplica
=== CONT  TestAccSecretsManagerSecret_basic
=== CONT  TestAccSecretsManagerSecret_basicReplica
--- PASS: TestAccSecretsManagerSecret_basic (13.70s)
--- PASS: TestAccSecretsManagerSecret_basicReplica (16.08s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/secretsmanager	39.064s
2024/12/18 14:59:04 Initializing Terraform AWS Provider...
=== RUN   TestAccSTSCallerIdentityDataSource_basic
=== PAUSE TestAccSTSCallerIdentityDataSource_basic
=== CONT  TestAccSTSCallerIdentityDataSource_basic
--- PASS: TestAccSTSCallerIdentityDataSource_basic (7.90s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sts	22.676s
```

File: docs/running-and-writing-acceptance-tests.md [hunk 1]
```diff
@@ -1149,12 +1149,12 @@ func sweepThings(region string) error {
...
   conn := client.ExampleConn(ctx)
-  sweepResources := make([]sweep.Sweepable, 0)
+  var sweepResources []sweep.Sweepable
   var errs *multierror.Error
 
-  input := &example.ListThingsInput{}
+  input := example.ListThingsInput{}
 
-  err = conn.ListThingsPages(input, func(page *example.ListThingsOutput, lastPage bool) bool {
+  err = conn.ListThingsPages(ctx, input, func(page *example.ListThingsOutput, lastPage bool) bool {
```
Comment (jar-b) [inline docs/running-and-writing-acceptance-tests.md:1157]:
```suggestion
  err = conn.ListThingsPages(ctx, &input, func(page *example.ListThingsOutput, lastPage bool) bool {
```

**Review by jar-b — DISMISSED** [review]
LGTM 🚀

**Comment by gdavison** [comment]
> Interestingly, this resulted in a 0.08% reduction in max RSS based on 3 runs with and without the changes. I just wanted to make sure it didn't increase but a little surprised at a reduction.

I didn't expect that much difference, since this change mostly affects tagging functions. Interesting…

**Review by johnsonaj — APPROVED** [review]
LGTM 🚀

**Pull request merged**